### PR TITLE
add ManagementContract.ListContracts

### PR DESCRIPTION
**Pull request by devhawk**
Add method to get an IEnumerable of all deployed contracts for use in blockchain toolkit

### CODE REVIEW

File: src/neo/SmartContract/Native/ManagementContract.cs [hunk 1]
```diff
@@ -44,6 +46,13 @@ internal override void OnPersist(ApplicationEngine engine)
             }
         }
 
+        static Lazy<byte[]> listContractsPrefix = new Lazy<byte[]>(() => new KeyBuilder(NativeContract.Management.Id, 8).ToArray());
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/ManagementContract.cs:49]: It doesn't need to be a `static` field.

**Review by shargon — APPROVED** [review]

File: src/neo/SmartContract/Native/ManagementContract.cs (hunk 1 above)
Comment (shargon) [reply src/neo/SmartContract/Native/ManagementContract.cs]: Done

**Review by shargon — APPROVED** [review]

File: src/neo/SmartContract/Native/ManagementContract.cs [hunk 2]
```diff
@@ -18,6 +20,7 @@ public sealed class ManagementContract : NativeContract
 
         private const byte Prefix_NextAvailableId = 15;
         private const byte Prefix_Contract = 8;
+        private Lazy<byte[]> listContractsPrefix = new Lazy<byte[]>(() => new KeyBuilder(NativeContract.Management.Id, Prefix_Contract).ToArray());
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/ManagementContract.cs:23]: If you move the initialization to the constructor, then `Lazy<T>` is not needed.

File: src/neo/SmartContract/Native/ManagementContract.cs [hunk 3]
```diff
@@ -49,7 +48,8 @@ internal override void OnPersist(ApplicationEngine engine)
 
         public IEnumerable<ContractState> ListContracts(StoreView snapshot)
         {
-            return snapshot.Storages.Find(listContractsPrefix.Value).Select(kvp => kvp.Value.GetInteroperable<ContractState>());
+            byte[] listContractsPrefix = new KeyBuilder(NativeContract.Management.Id, Prefix_Contract).ToArray();
```
Comment (erikzhang) [inline src/neo/SmartContract/Native/ManagementContract.cs:51]: Please use `CreateStorageKey()`.

Comment (shargon) [reply src/neo/SmartContract/Native/ManagementContract.cs]: Done

**Review by shargon — DISMISSED** [review]

**Comment by devhawk** [comment]
looks good

**Review by erikzhang — APPROVED** [review]

**Pull request merged**